base/linalg/triangular.jl

```diff
@@ -283,6 +283,155 @@ function -(A::UnitUpperTriangular)
     UpperTriangular(Anew)
 end
 
+# copy and scale
+function copy!(A::UpperTriangular,B::UpperTriangular)
+    copy!(A.data,B.data)
+end
+function copy!(A::UnitUpperTriangular,B::UnitUpperTriangular)
+    copy!(A.data,B.data)
+end
+function copy!(A::LowerTriangular,B::LowerTriangular)
+    copy!(A.data,B.data)
+end
+function copy!(A::UnitLowerTriangular,B::UnitLowerTriangular)
+    copy!(A.data,B.data)
+end
```

Comment:
Note that all of this duplication could be eliminated if the `copy!` method bodies are identical, simply by making a `copy!{T<:Union(...)}(A::T, B::T)` method that acts on the `Union` of the types.  (Or better yet, these types should maybe have a common abstract supertype.)